dist/pythonlibs/testrunner/pytest.py

```diff
@@ -20,17 +20,37 @@
 import subprocess
 import re
 import time
+import traceback
 
 import pexpect
 import pytest
 import testrunner
 
-
 DEFAULT_TIMEOUT = 10
 TEST_LOG_CONSOLE = bool(int(os.environ.get('TEST_LOG_CONSOLE', '1')))
 TEST_TIMEOUT_ATTR = 'TIMEOUT'
 
 
+class CustomSpawn(pexpect.spawn):
+    """Convenient subclass to better catch pexpect timeout and eof errors."""
+
+    def expect(self, pattern, timeout=-1, searchwindowsize=-1, async_=False,
+               **kw):
+        # pylint:disable=arguments-differ
+        try:
+            super(CustomSpawn, self).expect(pattern, timeout=timeout,
+                                            searchwindowsize=searchwindowsize,
+                                            async=async_, **kw)
+        except pexpect.TIMEOUT:
+            error = traceback.format_stack()[-2]
```

Comment:
I found this here: https://pexpect.readthedocs.io/en/stable/_modules/pexpect/exceptions.html would it make sense to use it to get the error ? it also reminds me what we are doing in `testrunner` (from memory only), so maybe replacing it also there would work. (Not currently tested)

For a version of this as a "NodeClass" wrapper, I would prefer to use normal exceptions here.
I would change the `pytest` specific handling in a specific subclass to separate concerns and still allow using the class outside of tests.